proposed/links.md

```diff
@@ -162,14 +181,92 @@ interface LinkInterface
 }
 ~~~
 
-#### 3.2.1 `Psr\Http\Link\LinkCollectionInterface`
+### 3.2 `Psr\Http\Link\EvolvableLinkInterface`
+
+~~~php
+<?php
+
+namespace Psr\Http\Link;
+
+/**
+ * An evolvable link value object.
+ */
+interface EvolvableLinkInterface extends LinkInterface
+{
+    /**
+     * Returns an instance with the specified href.
```

Comment:
The wording here is the same as in PSR-7. I copied that deliberately to handle the "same instance no-op" optimization that happened to PSR-7 late in the game.